src/PeriodEnd/PaymentsDue/SFA.DAS.ProviderPayments.Calc.PaymentsDue.UnitTests/PaymentsDueProcessor/WhenChangingAUlnForALearner.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using CS.Common.External.Interfaces;
+using MediatR;
+using Moq;
+using NLog;
+using NUnit.Framework;
+using SFA.DAS.Payments.DCFS.Context;
+using SFA.DAS.ProviderPayments.Calc.Common.Context;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.CollectionPeriods;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.CollectionPeriods.GetCurrentCollectionPeriodQuery;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.Earnings;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.Earnings.GetProviderEarningsQuery;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.Providers;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.Providers.GetProvidersQuery;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.RequiredPayments;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.RequiredPayments.AddRequiredPaymentsCommand;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.RequiredPayments.GetPaymentHistoryQuery;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Application.RequiredPayments.GetPaymentHistoryWhereNoEarningQuery;
+
+namespace SFA.DAS.ProviderPayments.Calc.PaymentsDue.UnitTests.PaymentsDueProcessor
+{
+    [TestFixture]
+    public class WhenChangingAUlnForALearner
+    {
+        private ILogger _logger;
+
+        private Mock<IMediator> _mediator;
+
+        private Mock<IExternalContext> _externalContext;
+
+        private PaymentsDue.PaymentsDueProcessor _processor;
+
+        [SetUp]
+        public void Setup()
+        {
+            _logger = Mock.Of<ILogger>();
+            _mediator = new Mock<IMediator>();
+            _externalContext = new Mock<IExternalContext>();
+            _externalContext.Setup(c => c.Properties)
+                .Returns(new Dictionary<string, string>
+                {
+                    { ContextPropertyKeys.TransientDatabaseConnectionString, "" },
+                    { ContextPropertyKeys.LogLevel, "DEBUG" },
+                    { PaymentsContextPropertyKeys.YearOfCollection, "1718" }
+                });
+
+            _processor = new PaymentsDue.PaymentsDueProcessor(_logger, _mediator.Object, new ContextWrapper(_externalContext.Object));
+        }
+
+        [Test]
+        public void ThenPaymentsAreStillMadeCorrectly()
```

Comment:
could we change the name of the test to say, "then no payments are due" or something like that ?